sherpa/astro/io/pyfits_backend.py

```diff
@@ -1465,3 +1465,100 @@ def set_arrays(filename, args, fields=None, ascii=True, clobber=False):
     hdu = fits.table_to_hdu(tbl)
     hdu.name = 'TABLE'
     hdu.writeto(filename, overwrite=True)
+
+
+def _add_header(hdu, header):
+    """Add the header items to the HDU.
+
+    Parameters
+    ----------
+    hdu : astropy HDU
+    header : list of sherpa.astro.io.xstable.HeaderItem
+    """
+
+    for hdr in header:
+        card = [hdr.name, hdr.value]
+        if hdr.desc is not None or hdr.unit is not None:
+            comment = "" if hdr.unit is None else f"[{hdr.unit}] "
+            comment += "" if hdr.desc is None else hdr.desc
```

Comment:
Since the previous line initializes comment, there is no need to append "" to comment when hdr,desc is None.